File: src/main/java/com/roger/student/Student.java
package com.roger.student;

public class Student {
    String id;
    String name;
    int english;
    int math;
    static int pass = 60;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){
//        int max = ;
        /*if (english > math){
            max = english;
        } else {
            max = math;
        }*/
        return (english > math) ? english : math;
    }

    public void print(){
        int average = getAverage();
        System.out.print(name + "\t" + english + "\t" + math + "\t" + getAverage()
        +"\t" + ((getAverage() >= pass)? "PASS" : "FAILED"));
        char grading = 'F';
        switch (average/10){
            case 10:
            case 9:
                grading ='A';
                break;
            case 8:
                grading ='B';
                break;
            case 7:
                grading ='C';
                break;
            case 6:
                grading ='D';
                break;
            default:
                grading = 'F';

        }
        System.out.println("\t" + grading);
/*        if (average >= 90 && average <=100){
            grading = 'A';
        }else if(average >= 80 && average <90){
            grading = 'B';
        }else if(average >= 70 && average <80){
            grading = 'C';
        }else if(average >= 60 && average <70){
            grading = 'D';
        }else{
            grading = 'F';
        }*/

/*        if (getAverage() >= 60){
            System.out.println("\tPASS");
        } else {
            System.out.println("\tFAILED");
        }*/
    }
    public int getAverage(){
        return (english+math)/2;
    }
}

File: src/main/kotlin/com/roger/kotlin/Student.kt
package com.roger.kotlin

import java.util.*

fun main() {
//    userInput()

    val stu = Student("Roger", 77, 99)
    val stu1 = Student("Eric", 30, 80)
    val stu2 = Student("Jane", 25, 60)
    val gstu = GraduateStudent("Jack", 55, 65, 60)
    gstu.print()
    stu.print()
    stu1.print()
    stu2.print()
    println("High score: ${stu.highest()}")
}

class GraduateStudent(name: String?, english: Int, math: Int,var thesis: Int ) : Student(name, english, math){
    companion object{
        var pass = 70
    }

    override fun print() {
        println("$name\t$english\t$math\t$thesis\t${getAverage()}\t${passOrFail()}\t${grading()}")
    }

    override fun passOrFail()= if(getAverage() >= pass)"PASS" else "FAILED"
}

open class Student (var name:String?, var english:Int, var math:Int) {
    companion object{
        @JvmStatic
        var pass:Int = 60
        fun test(){
            println("testing")
        }
    }
    open fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFail()}\t${grading()}")
    }
    open fun passOrFail() = if(getAverage() >= pass)"PASS" else "FAILED"

    fun grading() = when(getAverage()){
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
    }

    fun getAverage() = (english+math)/2

    fun nameCheck() = name?.length

    fun highest() = if (english > math){
        println ("english")
        english
    } else {
        println ("math")
        math
    }
/*        if (english > math){
            max = english
        } else {
            max = math
        }*/
}


private fun userInput() {
    val scanner = Scanner(System.`in`)
//    println("Please enter student's name:")
//    var name = scanner.next()
    var name = null
    println("Please enter student's english:")
    var english = scanner.nextInt()
    println("Please enter student's math:")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
    stu.nameCheck()
}
